src/CodeSigning/CodeSigning/Commands/GetAzureCodeSigningRootCert.cs

```diff
@@ -0,0 +1,127 @@
+        [ValidateNotNullOrEmpty()]
+        public string ProfileName { get; set; }
+        [Parameter(Mandatory = true,
+            Position = 2,
+            ParameterSetName = ByAccountProfileNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The endpoint url used to submit request to Azure CodeSigning.")]
+        public string EndpointUrl { get; set; }
+
+
+        /// <summary>
+        /// Metadata File Path
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = ByMetadataFileParameterSet,
+             ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Metadata File path. Cmdlet constructs the FQDN of an account profile based on the Metadata File and currently selected environment.")]
+        [ValidateNotNullOrEmpty]
+        public string MetadataFilePath { get; set; }
+
+        [Parameter(Mandatory = true,
+          Position = 3,
+          ParameterSetName = ByAccountProfileNameParameterSet,
+          ValueFromPipelineByPropertyName = true,
+          HelpMessage = "Downloaded Root Cert file full path, including file name")]
+        [Parameter(Mandatory = true,
+          Position = 3,
+          ParameterSetName = ByMetadataFileParameterSet,
+          ValueFromPipelineByPropertyName = true,
+          HelpMessage = "Downloaded Root Cert file full path, including file name")]
+        [ValidateNotNullOrEmpty]
+        public string Destination { get; set; }
+        #endregion
+
+        public override void ExecuteCmdlet()
+        {
+            Stream rootcert;
+
+            if (!string.IsNullOrEmpty(AccountName))
+            {
+                rootcert = CodeSigningServiceClient.GetCodeSigningRootCert(AccountName, ProfileName, EndpointUrl);
+                WriteRootCert(rootcert);
+            }
+            else if (!string.IsNullOrEmpty(MetadataFilePath))
+            {
+                rootcert = CodeSigningServiceClient.GetCodeSigningRootCert(MetadataFilePath);
+                WriteRootCert(rootcert);
+            }
+        }
+        private void WriteRootCert(Stream rootcert)
+        {   
+            var downloadPath = ResolvePath(Destination);
+
+            var fileStream = new FileStream(downloadPath, FileMode.Create, FileAccess.Write);
+            rootcert.CopyTo(fileStream);
+            fileStream.Dispose();
+
+            //read thumbprint and subject namme                      
+            byte[] rawData = File.ReadAllBytes(downloadPath);
+            X509Certificate2 x509 = new X509Certificate2(rawData);
+         
+            WriteObject(downloadPath.Replace("\\", @"\"));           
+
+            PSSigningCertificate pscert = new PSSigningCertificate();
+            pscert.Subject = x509.Subject;
+            pscert.Thumbprint = x509.Thumbprint;
+
+            WriteObject(pscert, false);
```

Comment:
Is it OK to print thumbprint to the console? User may run this in a public pipeline.

Reply:
Yes, the thumbprint is just an identifier that's results from a one-way hashing function, and its not considered private.